WordPress/gutenberg
# Enforce @since tags in /packages/block-serialization-default-parser/ and other files

## What?
This PR enhances support for checking `@since` tags within various code elements such as class, trait, interface, enum declarations, properties, and method declarations. Additionally, it extends support to check `@since` tags for WordPress functions that invoke hooks like `'do_action'`, `'do_action_ref_array'`, `'apply_filters'`, and `'apply_filters_ref_array'`.

The sniff only checks `/**` style comments.
The sniffs supports multiple `@since` tags per token.

Fixes https://github.com/WordPress/gutenberg/issues/59826.

## Why?
This PR improves the overall code adherence to WordPress coding standards. By enforcing `@since` tags, it ensures better documentation and version tracking, making it easier for developers to understand the evolution of the codebase over time.


## How?
- [x] Added support for checking `@since` tags for class, trait, interface, and enum declarations.
- [x] Added support for checking `@since` tags for class and trait properties;
- [x] Added support for checking `@since` tags for class, interface, and trait method declarations;
- [x] Added support for checking `@since` tags for WordPress functions that invoke hooks: `do_action()`, `do_action_ref_array()`, `apply_filters()`, `apply_filters_ref_array()`.
- [x] Refactored unit tests; duplicate code moved to the new `GutenbergCS\Gutenberg\Tests\AbstractSniffUnitTest` class.

## Testing Instructions

1. Define a class, trait, interface, and enum without a `@since` tag and ensure that the linter checks fail.
2. Define a function, class method, interface method, and trait method without a `@since` tag and ensure that the linter checks fail. Ensure that the sniff skips checking private methods.
3. Attempt to use one of the WordPress functions that invoke hooks such as `do_action()`, `do_action_ref_array()`, `apply_filters()`, `apply_filters_ref_array()` without specifying a `@since` tag. Confirm that the linter checks fail.
4. Define a class property or trait property without a `@since` tag and ensure that the linter checks fail.

### Testing Instructions for Keyboard


## Screenshots or screencast
